src/Hl7.Fhir.Serialization.Shared.Tests/RoundtripNewSerializers.cs

```diff
@@ -0,0 +1,245 @@
+        private static readonly string intermediate1Folder = "intermediate1";
+        private static readonly string intermediate2Folder = "intermediate2";
+
+        [DynamicData(nameof(prepareExampleZipFilesXml), DynamicDataSourceType.Method, DynamicDataDisplayName = nameof(GetTestDisplayNames))]
+        [DataTestMethod]
+        [TestCategory("LongRunner")]
+        public void FullRoundtripOfAllExamplesXmlNewSerializer(string file, string baseTestPath, FhirXmlPocoSerializer xmlSerializer, FhirXmlPocoDeserializer xmlDeserializer, JsonSerializerOptions jsonOptions)
+        {
+            doRoundTrip(baseTestPath, file, xmlSerializer, xmlDeserializer, jsonOptions);
+        }
+
+        [DynamicData(nameof(prepareExampleZipFilesJson), DynamicDataSourceType.Method, DynamicDataDisplayName = nameof(GetTestDisplayNames))]
+        [DataTestMethod]
+        [TestCategory("LongRunner")]
+        public void FullRoundtripOfAllExamplesJsonNewSerializer(string file, string baseTestPath, FhirXmlPocoSerializer xmlSerializer, FhirXmlPocoDeserializer xmlDeserializer, JsonSerializerOptions jsonOptions)
+        {
+            doRoundTrip(baseTestPath, file, xmlSerializer, xmlDeserializer, jsonOptions);
+        }
+
+        private static IEnumerable<object[]> prepareExampleZipFilesXml()
+        {
+            return prepareExampleZipFiles("examples.zip", Path.Combine(Path.GetTempPath(), xmlTestFolder));
+        }
+
+        private static IEnumerable<object[]> prepareExampleZipFilesJson()
+        {
+
+            return prepareExampleZipFiles("examples-json.zip", Path.Combine(Path.GetTempPath(), jsonTestFolder));
+        }
+
+        private static IEnumerable<object[]> prepareExampleZipFiles(string zipname, string targetDir)
+        {
+            ZipArchive examples = extractTestZip(zipname);
+            createEmptyDir(targetDir);
+
+            var inputPath = Path.Combine(targetDir, inputFolder);
+            createEmptyDir(inputPath);
+
+            examples.ExtractToDirectory(inputPath);
+            var files = getFiles(Path.Combine(inputPath), new[] { "*.xml", "*.json" }, SearchOption.AllDirectories).ToList();
+            var objects = new List<object[]>();
+
+            var intermediate1Path = Path.Combine(targetDir, intermediate1Folder);
+            createEmptyDir(intermediate1Path);
+
+            var intermediate2Path = Path.Combine(targetDir, intermediate2Folder);
+            createEmptyDir(intermediate2Path);
+
+            var xmlSerializer = new FhirXmlPocoSerializer(ModelInfo.ModelInspector.FhirRelease);
+            var xmlDeserializer = new FhirXmlPocoDeserializer(ModelInfo.ModelInspector);
+            var jsonOptions = new JsonSerializerOptions().ForFhir(ModelInfo.ModelInspector).Pretty();
+
+            files.Where(f => !skipFile(f)).ToList()
+                 .ForEach(f => objects.Add(new object[] { f, targetDir, xmlSerializer, xmlDeserializer, jsonOptions }));
```

Comment:
Since `objects` is empty, you can just use `.Select` here, and the `ToList()` at then end. No need for a temp `objects` here!